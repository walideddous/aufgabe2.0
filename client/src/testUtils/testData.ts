const table = () => {
  let tab =[];
  for(let i=0; i<=16; i++){
    tab.push({
      angle: -83.19868517601843,
      coord: { WGS84: { lat: 46.17857, lon: 6.08606 } },
      distance: 0.17031643195093935,
      modes: ["4", "5"],
      name: "Bernex 1",
      _id: "5f6206390d5658001cd959b4" + i,
    });  
  }
  return tab;
};

export const stateDND = {
  suggestions: {
    title: "Suggestion",
    items: table(),
  },
  trajekt: {
    title: "Stop sequence",
    items: [
      {
        coord: { WGS84: { lat: 46.17831, lon: 6.08824 } },
        index: 0,
        modes: ["4", "5"],
        name: "Confignon, croisée",
        _id: "5f6205c60d5658001cd9480b",
      },
    ],
  },
};

export const selected = {
  coord: { WGS84: { lat: 46.17831, lon: 6.08824 } },
  index: 0,
  modes: ["4", "5"],
  name: "Confignon, croisée",
  _id: "5f6205c60d5658001cd9480b",
};

export const stopSequenceList = [
  {
      _id: "638b7703-d171-4e5f-8e78-21258f75cee4",
      name: "asdsadsa",
      modes: "4",
      schedule: [
          {
              date: "2020.11.24-2020.12.16",
              dayTime: [
                  {
                      day: [
                          "Mon"
                      ],
                      time: [
                          "05:00",
                          "11:00"
                      ]
                  }
              ]
          }
      ],
      stopSequence: [
          {
              _id: "5f62060e0d5658001cd95358",
              name: "Schlieren, Gasometerbrücke",
              modes: [
                  "4",
                  "5"
              ],
              coord: {
                  WGS84: {
                      lat: 47.39761,
                      lon: 8.4606
                  }
              }
          }
      ]
  }
]

export const currentStopSequence =   {
  _id: "638b7703-d171-4e5f-8e78-21258f75cee4",
  name: "asdsadsa",
  modes: "4",
  schedule: [
      {
          date: "2020.11.24-2020.12.16",
          dayTime: [
              {
                  day: [
                      "Mon"
                  ],
                  time: [
                      "05:00",
                      "11:00"
                  ]
              }
          ]
      }
  ],
  stopSequence: [
      {
          _id: "5f62060e0d5658001cd95358",
          name: "Schlieren, Gasometerbrücke",
          modes: [
              "4",
              "5"
          ],
          coord: {
              WGS84: {
                  lat: 47.39761,
                  lon: 8.4606
              }
          }
      }
  ]
}

export const distance = [
  {
    angle: -95.16982788755787,
    distance: 0.163130219415872,
    from: "Camedo, Centovalli",
    to: {
      coord: { WGS84: { lat: 46.17831, lon: 6.08824 } },
      modes: ["4", "5"],
      name: "Confignon, croisée",
      _id: "5f6205c60d5658001cd9480b",
    },
  },
];

export const stations = [
  {
    _id: "5f6203bb0d5658001cd8f85a",
    name: "Basel",
    coord: {
      WGS84: {
        lat: 47.54741,
        lon: 7.58956,
      },
    },
    modes: [],
  },
  {
    _id: "5f6203bb0d5658001cd8f85b",
    name: "Lyon",
    coord: {
      WGS84: {
        lat: 45.74506,
        lon: 4.84184,
      },
    },
    modes: [],
  },
];
