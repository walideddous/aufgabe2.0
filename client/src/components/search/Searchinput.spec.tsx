import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Searchinput from "./SearchInput";
// Import the data to test With
import { stations } from "../../testUtils/testData";
import { act } from "@testing-library/react";

const makeProps = (props: any) => ({
  stations: [],
  handleSelectAutoSearch() {},
  ...props,
});

const setUp = (props: any) => {
  const component = shallow(<Searchinput {...props} />);
  return component;
};

describe("Searchinput component", () => {
  let wrappedComponent: any;

  const handleSelectAutoSearch = jest.fn();

  beforeEach(() => {
    wrappedComponent = setUp(
      makeProps({
        stations,
        handleSelectAutoSearch,
      })
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("Should test the jest framework", () => {
    expect(true).toBe(true);
  });

  it("Should match snapShot with the Seachinput component", () => {
    expect(toJSON(wrappedComponent)).toMatchSnapshot();
  });

  it("Should test the AutoComplete component", () => {
    const searchField = wrappedComponent.find("#stops_autoComplete");

    act(() => {
      searchField.props().onSearch("Basel");
      searchField.props().onSelect("Basel", { stop: { name: "Basel" } });
    });

    expect(searchField.props().value).toBe("");
    expect(handleSelectAutoSearch).toBeCalled();
  });
});
