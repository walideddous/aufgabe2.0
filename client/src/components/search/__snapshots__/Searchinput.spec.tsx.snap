// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchinput component Should match snapShot with the Seachinput component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "start",
      "left": "18%",
      "position": "absolute",
      "top": "25px",
      "width": "70%",
      "zIndex": 10,
    }
  }
>
  <ForwardRef(AutoComplete)
    id="stops_autoComplete"
    onSearch={[Function]}
    onSelect={[Function]}
    open={false}
    placeholder="Haltestellenname eingeben"
    style={
      Object {
        "width": "100%",
      }
    }
    value=""
  >
    <Option
      key="5f6203bb0d5658001cd8f85a"
      stop={
        Object {
          "_id": "5f6203bb0d5658001cd8f85a",
          "coord": Array [
            47.54741,
            7.58956,
          ],
          "modes": Array [],
          "name": "Basel",
        }
      }
      value="Basel"
    >
      Basel
    </Option>
    <Option
      key="5f6203bb0d5658001cd8f85b"
      stop={
        Object {
          "_id": "5f6203bb0d5658001cd8f85b",
          "coord": Array [
            45.74506,
            4.84184,
          ],
          "modes": Array [],
          "name": "Lyon",
        }
      }
      value="Lyon"
    >
      Lyon
    </Option>
  </ForwardRef(AutoComplete)>
</div>
`;
